_rules/SC2-4-2-page-has-title.md

```diff
@@ -22,19 +22,19 @@ authors:
 
 ### Applicability
 
-The rule applies to any page where the root element is an `html` element, and which is not embedded in another page.
+The rule applies to any page where the [document element](https://www.w3.org/TR/dom/#document-element) is an `html` element, and where the page is not embedded in another page.
 
-**Note**: Pages may be embedded inside other pages through elements such as iframes and object elements.
+**Note**: Pages may be embedded inside other pages through elements such as `iframe` and `object` elements.
 
 ### Expectation 1
 
-The page contains at least one `title` element.
+The [document element](https://www.w3.org/TR/dom/#document-element) has at least one [descendant](https://www.w3.org/TR/dom41/#concept-tree-descendant) that is an HTML `title` element.
 
-**Note**: The `title` element exists in other namespaces such as SVG. These are not `title` elements for HTML document and should be ignored.
+**Note**: The `title` element exists in other namespaces such as SVG. These are not HTML `title` elements and should be ignored for this rule.
 
 ### Expectation 2
 
-The first `title` element contains [non-empty text](#non-empty).
+The first `title` element contains [text nodes](https://www.w3.org/TR/dom/#text) that do not only consist of [Unicode separator characters](https://www.unicode.org/versions/Unicode11.0.0/ch04.pdf#G134153).
```

Comment:
Can we simplify this a little by doing:

> The first `title` element contains [text](https://www.w3.org/TR/dom/#text) that does not only consist of [whitespace (specifically, Unicode separator characters)](https://www.unicode.org/versions/Unicode11.0.0/ch04.pdf#G134153).

Reply:
@WilcoFiers, we don't find your suggestion to be as unambiguous as we would like.

Reply:
Which part of it? Alternative proposal:

> ... contains [text content](https://www.w3.org/TR/dom41/#dom-node-textcontent) that does not only consist of [Unicode separator characters](https://www.unicode.org/versions/Unicode11.0.0/ch04.pdf#G134153) (i.e. space, line breaks and other whitespace characters)

Text content is what the HTML 5.2 spec says needs to be used to derive the document title from the title element, so it's the right technical term, and it reads better than "text nodes" IMO.

Reply:
HTML defines "child text content" as a term (https://www.w3.org/TR/html/infrastructure.html#child-text-content), not "text content". In general, I don't really think it's appropriate to refer do DOM attributes in the manner proposed, e.g. use "text content" to mean `node.textContent` or "document title" to mean `document.title`. If referring to DOM attributes or properties, be explicit about it.

Reply:
Fair point. But then shouldn't this simply be using `document.title`?

Reply:
I don't think it should, nor do I think that rules in general should refer to specific APIs, be they DOM or something else. That's an implementation detail and rules shouldn't dictate the "how" but rather the "what". The "what" in this this case is the concatenation of the data of the text nodes that are children of the `<title>` element, i.e. the "child text content". The "how" _could_ be `document.title`.

Reply:
I don't mean to say to literally access the document.title property. You're right, that's clearly implementation. I mean using "document title", linking to the algorithm for deriving it, the same way we talk about "accessible name" and referencing that algorithm.

Reply:
That makes perfect sense and for that very same reason I'd also suggest using the term "child text content" similar to how we use the term "accessible name". We could even add "document title" to the glossary and define it as "the child text content of the first `<title>` element that is a descendant of the document".